RestaurantReservation.API/Base/AppControllerBase.cs

```diff
@@ -0,0 +1,39 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservation.Core.Bases;
+using System.Net;
+
+namespace RestaurantReservation.API.Base;
+
+[ApiController]
+public class AppControllerBase : ControllerBase
+{
+    private IMediator _mediatorInstance;
+    protected IMediator Mediator => _mediatorInstance ??= HttpContext.RequestServices.GetService<IMediator>();
+
+    #region Actions
+    public ObjectResult NewResult<T>(Response<T> response)
```

Comment:
I prefer not using this, try to avoid it and use the result you want on its method